src/thor/route_action.cc

```diff
@@ -60,23 +60,58 @@ void via_discontinuity(
     return;
   }
 
-  // Add a discontinuity if the edges are opposing
-  GraphId in_edge_id(in_pe->graph_id());
-  GraphId out_edge_id(out_pe->graph_id());
-  GraphId opp_edge_id = reader.GetOpposingEdgeId(in_edge_id);
-  if (opp_edge_id == out_edge_id) {
-    PointLL snap_ll(in_pe->ll().lng(), in_pe->ll().lat());
-    double dist_along = in_pe->percent_along();
-
-    // Insert a discontinuity so the last edge of the first segment is trimmed at the beginning
-    // from 0 to dist_along. Set the first
-    vias.insert(
-        {path_index + (flip_index ? 1 : 0), {{false, PointLL(), 0.0}, {true, snap_ll, dist_along}}});
-
-    // Insert a second discontinuity so the next (opposing) edge is trimmed at the end from
-    // 1-dist along to 1
-    vias.insert({path_index + (flip_index ? 0 : 1),
-                 {{true, snap_ll, 1.0 - dist_along}, {false, PointLL(), 1.0}}});
+  PointLL snap_ll(in_pe->ll().lng(), in_pe->ll().lat());
+  double dist_along = in_pe->percent_along();
+
+  // Special case : When you have a single edge route, and you have multiple throughs/vias
+  // along it. In this case you need to trim both sides of an edge before you add it if you are
+  // continuing on the same edge.
+  /*if (loc.path_edges_size() == 1 && (std::next(loc.path_edges().begin()) != loc.path_edges().end()))
+  {
+          // Insert a discontinuity so the last edge of the first segment is trimmed at the beginning
+  & end
+          // from 0 to dist_along. Set the first
+          intermediate_locs.insert({path_index + (flip_index ? 1 : 0),
+                                   {{true, PointLL(), 0.0, path_index}, {true, snap_ll, dist_along,
+  path_index}}});
+
+          // Insert a second discontinuity so the next (opposing) edge is trimmed at the beginning &
+  end from
+          // 1-dist along to 1
+          intermediate_locs.insert({path_index + (flip_index ? 0 : 1),
+                                   {{true, snap_ll, 1.0 - dist_along, path_index}, {true,
+  PointLL(), 1.0, path_index}}}); } else {*/
+  // Insert a discontinuity so the last edge of the first segment is trimmed at the beginning
+  // from 0 to dist_along. Set the first
+  intermediate_locs.insert(
+      {path_index + (flip_index ? 1 : 0),
+       {{false, PointLL(), 0.0, path_index}, {true, snap_ll, dist_along, path_index}}});
+
+  // Insert a second discontinuity so the next (opposing) edge is trimmed at the end from
+  // 1-dist along to 1
+  intermediate_locs.insert(
+      {path_index + (flip_index ? 0 : 1),
+       {{true, snap_ll, 1.0 - dist_along, path_index}, {false, PointLL(), 1.0, path_index}}});
+  //}
+  for (auto& intermediate_loc : intermediate_locs) {
```

Comment:
this will be quite repetative. why wouldnt you just print the last two you added.  you can do this by assigning the result of `insert` above to a variable and printing the stuff that comes out of the iterator